### PR TITLE
Channel Mutes & Cleanup Query Options

### LINKED ISSUE #74: [BUG] QueryChannel pagination limit incorrect implementation
Hi people

According to the source code, pagination parameters for the channel querying should be passed within the `filter_conditions`. 
In docs it is suggested to pass the query parameters on the root level of the payload JSON structure (`queryChannelRequest` struct in Go SDK instead of `QueryOption` nested struct).

**References**
In Go SDK:
https://github.com/GetStream/stream-chat-go/blob/master/query.go#L102

In docs:
https://getstream.io/chat/docs/query_channels/#query-options

**Steps to reproduce**
1. Create at least two channels
2. Call `QueryChannels` method without any pagination options passed
3. Make sure you receive your channels in the response
4. Call `QueryChannels` method with explicit pagination rules (limit: 10, offset: 1)

_Expected_
Channels are returned respecting the provided pagination parameters

_Actual_
Empty channels list is returned

### PR DESCRIPTION
**Pull request by keyneston**
This is a fork of #60 as @thesyncim had already started down this path. I just took it a step further and did some additional cleanup of query options.

This should also fix #74

### CODE REVIEW

**Comment by keyneston** [comment]
Rebased onto master.

File: channel.go [hunk 1]
```diff
@@ -485,6 +485,41 @@ func (ch *Channel) RejectInvite(userID string, message *Message) error {
...
+
+	data := map[string]interface{}{
+		"user_id":     userID,
+		"channel_cid": ch.CID,
+	}
+	if expiration != nil {
+		data["expiration"] = int(expiration.Seconds())
+	}
+
+	var mute = &ChannelMuteResponse{}
```
Comment (ferhatelmas) [inline channel.go:501]: tiny but prefer `:=` here

reasoning: https://dave.cheney.net/practical-go/presentations/qcon-china.html#_use_a_consistent_declaration_style

File: user.go [hunk 2]
```diff
@@ -13,13 +13,27 @@ import (
...
+type ChannelMute struct {
+	User      User       `json:"user"`
+	Channel   Channel    `json:"channel"`
+	Expires   *time.Time `json:"expires"`
+	CreatedAt time.Time  `json:"created_at"`
+	UpdatedAt time.Time  `json:"updated_at"`
+}
+
+type ChannelMuteResponse struct {
+	ChannelMutes ChannelMute `json:"channel_mute"`
```
Comment (ferhatelmas) [inline user.go:34]: this is a single object so `ChannelMute` would be more correct

**Comment by keyneston** [comment]
@ferhatelmas I think I fixed all the things you pointed out. Can you reapprove?

**Pull request merged**